recipes-devtools/nsight-systems/nsight-systems-cli_2020.5.3.17-1.bb

```diff
@@ -0,0 +1,53 @@
+SUMMARY = "NVIDIA Nsight Systems is a multi-core CPU sampling profiler that \
+    provides an interactive view of captured profiling data, helping improve \
+    overall application performance."
+HOMEPAGE = "https://developer.nvidia.com/nsight-systems"
+LICENSE = "Proprietary"
+LIC_FILES_CHKSUM = " \
+    file://opt/nvidia/nsight-systems-cli/2020.5.3/NVIDIA_SLA.pdf;md5=f7646eea4f29c4f708ea0aa5bc6778a6 \
+"
+
+inherit nvidia_devnet_downloads
+
+SRC_URI = "${NVIDIA_DEVNET_MIRROR}/nsight-systems-cli-2020.5.3_${PV}_arm64.deb;subdir=nsight-systems-cli"
+SRC_URI[sha256sum] = "8a1d1bc76ee4a88793e1df26214e0df9aa69e0b435f9e79acdca84a41e7b3c8f"
+
+S = "${WORKDIR}/nsight-systems-cli"
+B = "${S}"
+
+COMPATIBLE_MACHINE = "tegra"
+
+do_configure() {
+    :
+}
+
+do_compile() {
+    :
+}
+
+do_install() {
+    install -d ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/target-linux-armv8
+    cp -R --preserve=mode,timestamps,links --no-dereference ${B}/opt/nvidia/nsight-systems-cli/2020.5.3/target-linux-armv8/* ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/target-linux-armv8/
+    install -d ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8
+    cp -R --preserve=mode,timestamps,links --no-dereference ${B}/opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8/* ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8/
+    install -d ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/bin
+    cp -R --preserve=mode,timestamps,links --no-dereference ${B}/opt/nvidia/nsight-systems-cli/2020.5.3/bin/* ${D}/opt/nvidia/nsight-systems-cli/2020.5.3/bin/
+}
+
+INHIBIT_PACKAGE_STRIP = "1"
+INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
+INHIBIT_SYSROOT_STRIP = "1"
+
+FILES_${PN} = " \
+    /opt/nvidia/nsight-systems-cli/2020.5.3/target-linux-armv8 \
+    /opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8 \
+    /opt/nvidia/nsight-systems-cli/2020.5.3/bin \
+"
+INSANE_SKIP_${PN} = "ldflags"
+
+PRIVATE_LIBS = " \
+    libnvlog.so \
+    libstdc++.so.6 \
```

Comment:
Ouch... the package includes its own copy of `libstdc++`?  Is it really needed?

Reply:
It is not really needed tbh,
There is a tool `QdstrmImporter` taht you can find here `/opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8`
which used to convert files from format `.qdstrm` to `.qdrep`.
For more details, please refer to the [Nvidia documentation](https://docs.nvidia.com/nsight-systems/2020.3/profiling/index.html#importing-qdstrm-files)

Reply:
Maybe this should be split into a different package?

Reply:
Separate packages for the 'host' tools vs the 'target' sounds like a good idea.

Looking at the package contents, it looks like the target-side directory also contains a bunch of libraries (as well as sqlite3 binary and python interpreter).  I think this recipe is a good candidate for just setting `EXCLUDE_FROM_SHLIBS = "1"` to skip shared library processing for the entire set.

Reply:
`host-linux-armv8` is a part of the package `nsight-systems-cli`, it is just a name that nvidia provide.
All libraries and tool are dedicated to be executed in the target.

Reply:
That's understood, but you said these so-called host tools don't need to be present for the rest of the package to work. So it makes sense to have this recipe create multiple packages.

Reply:
Agree, I will create a separate package that contains `/opt/nvidia/nsight-systems-cli/2020.5.3/host-linux-armv8`